server/src/test/java/org/cloudfoundry/identity/uaa/oauth/AntPathRedirectResolverTests.java

```diff
@@ -46,147 +55,169 @@
-            assertTrue(resolver.redirectMatches("http://subdomain.domain.com", clientRedirectUri));
-            assertTrue(resolver.redirectMatches("http://another-subdomain.domain.com", clientRedirectUri));
-            assertTrue(resolver.redirectMatches("http://one.two.domain.com", clientRedirectUri));
+    @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+    enum RequestedRedirectUri {
+
+        eg01("http://subdomain.domain.com",                 ____http, domain.parts(3), ____part_of_domain_dot_com, path.segments(0)),
+        eg02("http://another-subdomain.domain.com",         ____http, domain.parts(3), ____part_of_domain_dot_com, path.segments(0)),
+        eg03("http://one.two.domain.com",                   ____http, domain.parts(4), ____part_of_domain_dot_com, path.segments(0)),
+        eg04("http://domain.com/one",                       ____http, domain.parts(2), ____part_of_domain_dot_com, path.segments(1)),
+        eg05("http://domain.com/another",                   ____http, domain.parts(2), ____part_of_domain_dot_com, path.segments(1)),
+        eg06("http://domain.com/one/two",                   ____http, domain.parts(2), ____part_of_domain_dot_com, path.segments(2)),
+        eg07("http://subdomain.domain.com/one",             ____http, domain.parts(3), ____part_of_domain_dot_com, path.segments(1)),
+        eg08("http://subdomain.domain.com/another",         ____http, domain.parts(3), ____part_of_domain_dot_com, path.segments(1)),
+        eg09("http://subdomain.domain.com/one/two",         ____http, domain.parts(3), ____part_of_domain_dot_com, path.segments(2)),
+        eg10("http://another-subdomain.domain.com/one",     ____http, domain.parts(3), ____part_of_domain_dot_com, path.segments(1)),
+        eg11("http://another-subdomain.domain.com/another", ____http, domain.parts(3), ____part_of_domain_dot_com, path.segments(1)),
+        eg12("http://another-subdomain.domain.com/one/two", ____http, domain.parts(3), ____part_of_domain_dot_com, path.segments(2)),
+        eg13("http://one.two.domain.com/one",               ____http, domain.parts(4), ____part_of_domain_dot_com, path.segments(1)),
+        eg14("http://one.two.domain.com/another",           ____http, domain.parts(4), ____part_of_domain_dot_com, path.segments(1)),
+        eg15("http://one.two.domain.com/one/two",           ____http, domain.parts(4), ____part_of_domain_dot_com, path.segments(2)),
+        eg16("http://other-domain.com",                     ____http, domain.parts(2), not_part_of_domain_dot_com, path.segments(0)),
+        eg17("http://domain.io",                            ____http, domain.parts(2), not_part_of_domain_dot_com, path.segments(0)),
+        eg18("https://domain.com",                          not_http, domain.parts(2), ____part_of_domain_dot_com, path.segments(0)),
+        eg19("ws://domain.com",                             not_http, domain.parts(2), ____part_of_domain_dot_com, path.segments(0)),
+        ;
+
```

Comment:
Initially I envisaged just modifying / adding tests to mitigate future conflicts.
However, I needed to add a new set of tests (*.domain.com) that do allow subdomain matching. Copying and adapting the existing inner classes would have lead to code duplication. Instead, by grouping all test cases into a single enum, we can be sure that we won't miss any of them in each of the (currently) 4 scenarios.
An added benefit is that if a case fails, we'll know immediately which case failed from the test name, because each test case runs as an individual test. (Currently multiple test cases are bundled into a single method - if a test fails, its harder to figure out which test case is the faulty one.)